src/solvers/FrankWolfe.jl

```diff
@@ -60,18 +60,18 @@ mutable struct FrankWolfeState{
     ) where {
         P,
         Pr,
-        Op,
         T,
         TStop<:StoppingCriterion,
         TStep<:Stepsize,
         TM<:AbstractRetractionMethod,
         ITM<:AbstractInverseRetractionMethod,
     }
-        return new{P,T,Pr,Op,TStep,TStop,TM,ITM}(
+        sub_state_storage = maybe_wrap_evaluation_type(sub_state)
+        return new{P,T,Pr,typeof(sub_state_storage),TStep,TStop,TM,ITM}(
             p,
             initial_vector,
             sub_problem,
-            sub_state,
+            sub_state_storage,
```

Comment:
Why this renaming? The sub state is usually always the field `sub_state`, so this seems a bit unusual compared to the rest.

Reply:
I've just made a new name for the temporary variable to not have to rely on compiler figuring out how to optimize a variable changing its type.

Reply:
Ah, I misread that as being the new name of the field. If one reads that correctly this is of course fine.